Failure when executing generateDependencyGraph task

When I run `./gradlew generateDependencyGraph` in a project the build is failing if I use version 0.2.0 (see stacktrace). Switching to version 0.1.0 solves the problem. This seems somehow related to the command `command`.

Environment:
```
./gradlew --version                                                                                                                                                                                                        

------------------------------------------------------------
Gradle 4.6
------------------------------------------------------------

Build time:   2018-02-28 13:36:36 UTC
Revision:     8fa6ce7945b640e6168488e4417f9bb96e4ab46c

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          1.8.0_151 (Oracle Corporation 25.151-b12)
OS:           Linux 4.9.0-6-amd64 amd64
```

<details>
  <summary>Stacktrace (click to expand)</summary>

```
FAILURE: Build failed with an exception.
 
 * What went wrong:
 Execution failed for task ':generateDependencyGraph'.
 > A problem occurred starting process 'command 'command''
 
 * Try:
  Run with --scan to get full insights.

 
 * Exception is:
 org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':generateDependencyGraph'.
   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:103)
   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
   at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
   at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
   at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
   at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
   at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
   at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
   at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
   at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
   at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
   at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
   at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
   at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
   at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:256)
   at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
   at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
   at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
   at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
   at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:249)
   at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:238)
   at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
   at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
   at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
   at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
   at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:663)
   at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:597)
   at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
   at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
   at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
   at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
 Caused by: org.gradle.process.internal.ExecException: A problem occurred starting process 'command 'command''
   at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:226)
   at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:204)
   at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:349)
   at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:85)
   at org.gradle.internal.operations.BuildOperationIdentifierPreservingRunnable.run(BuildOperationIdentifierPreservingRunnable.java:39)
   ... 3 more
 Caused by: net.rubygrapefruit.platform.NativeException: Could not start 'command'
   at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27)
   at net.rubygrapefruit.platform.internal.WrapperProcessLauncher.start(WrapperProcessLauncher.java:36)
   at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:67)
   ... 4 more
 Caused by: java.io.IOException: Cannot run program "command" (in directory "/ramtmp/gradle_test"): error=2, No such file or directory
   at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
   ... 6 more
 Caused by: java.io.IOException: error=2, No such file or directory
   ... 7 more
```
</details>

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,6 +3,7 @@ language: android
 jdk: oraclejdk8
 
 before_install:
+  - brew install graphviz
   - pip install --user codecov
   # Download SDK
   - yes | sdkmanager "tools" &>/dev/null

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,7 +3,7 @@ language: android
 jdk: oraclejdk8
 
 before_install:
-  - brew install graphviz
+  - yes | sudo apt-get install graphviz
   - pip install --user codecov
   # Download SDK
   - yes | sdkmanager "tools" &>/dev/null

--- a/build.gradle
+++ b/build.gradle
@@ -63,6 +63,19 @@ repositories {
   gradlePluginPortal()
 }
 
+// Write the plugin's classpath to a file to share with the tests
+task createClasspathManifest {
+  def outputDir = file("$buildDir/$name")
+
+  inputs.files sourceSets.main.runtimeClasspath
+  outputs.dir outputDir
+
+  doLast {
+    outputDir.mkdirs()
+    file("$outputDir/plugin-classpath.txt").text = sourceSets.main.runtimeClasspath.join("\n")
+  }
+}
+
 dependencies {
   implementation localGroovy()
   implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion"
@@ -72,6 +85,7 @@ dependencies {
   testImplementation 'org.assertj:assertj-core:3.9.1'
   testImplementation 'com.android.tools.build:gradle:3.0.1'
   testImplementation "org.jetbrains.kotlin:kotlin-reflect:$kotlinVersion"
+  testRuntime files(createClasspathManifest)
 }
 
 sourceCompatibility = JavaVersion.VERSION_1_7

--- a/build.gradle
+++ b/build.gradle
@@ -63,7 +63,7 @@ repositories {
   gradlePluginPortal()
 }
 
-// Write the plugin's classpath to a file to share with the tests
+// Write the plugin's classpath to a file to share with the tests.
 task createClasspathManifest {
   def outputDir = file("$buildDir/$name")
 
@@ -104,6 +104,16 @@ pluginBundle {
       tags = ['gradle', 'graph', 'dependencies', 'graphviz', 'dot']
       description = POM_DESCRIPTION
     }
+  }
+}
+
+tasks.withType(Test) {
+  testLogging {
+    events "failed"
+    exceptionFormat "full"
+    showExceptions true
+    showStackTraces true
+    showCauses true
   }
 }
 

--- a/src/main/kotlin/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorTask.kt
+++ b/src/main/kotlin/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorTask.kt
@@ -20,8 +20,8 @@ open class DependencyGraphGeneratorTask : DefaultTask() {
   @TaskAction fun run() {
     val result = project.exec {
       it.standardOutput = NullOutputStream // Don't print the output.
-      it.executable = "command"
-      it.args("-v", "dot")
+      it.executable = "type"
+      it.args("dot")
       it.isIgnoreExitValue = true
     }
 

--- a/src/main/kotlin/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorTask.kt
+++ b/src/main/kotlin/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorTask.kt
@@ -18,10 +18,15 @@ open class DependencyGraphGeneratorTask : DefaultTask() {
   @OutputFile lateinit var outputFileImage: File
 
   @TaskAction fun run() {
+    val commandLine = when {
+      Os.isFamily(Os.FAMILY_MAC) -> listOf("command", "-v", "dot")
+      else -> listOf("sh", "-c", "command -v dot")
+    }
+
     val result = project.exec {
       it.standardOutput = NullOutputStream // Consume the output.
       it.errorOutput = NullOutputStream // Consume the output.
-      it.commandLine("sh", "-c", "command -v dot")
+      it.commandLine(commandLine)
       it.isIgnoreExitValue = true
     }
 

--- a/src/main/kotlin/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorTask.kt
+++ b/src/main/kotlin/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorTask.kt
@@ -19,9 +19,9 @@ open class DependencyGraphGeneratorTask : DefaultTask() {
 
   @TaskAction fun run() {
     val result = project.exec {
-      it.standardOutput = NullOutputStream // Don't print the output.
-      it.executable = "type"
-      it.args("dot")
+      it.standardOutput = NullOutputStream // Consume the output.
+      it.errorOutput = NullOutputStream // Consume the output.
+      it.commandLine("type", "dot")
       it.isIgnoreExitValue = true
     }
 

--- a/src/main/kotlin/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorTask.kt
+++ b/src/main/kotlin/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorTask.kt
@@ -21,7 +21,7 @@ open class DependencyGraphGeneratorTask : DefaultTask() {
     val result = project.exec {
       it.standardOutput = NullOutputStream // Consume the output.
       it.errorOutput = NullOutputStream // Consume the output.
-      it.commandLine("type", "dot")
+      it.commandLine("sh", "-c", "command -v dot")
       it.isIgnoreExitValue = true
     }
 

--- a/src/test/java/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorPluginTest.kt
+++ b/src/test/java/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorPluginTest.kt
@@ -5,11 +5,17 @@ import org.assertj.core.api.Java6Assertions.assertThat
 import org.gradle.api.internal.project.DefaultProject
 import org.gradle.api.plugins.JavaLibraryPlugin
 import org.gradle.testfixtures.ProjectBuilder
+import org.gradle.testkit.runner.GradleRunner
 import org.junit.Before
+import org.junit.Rule
 import org.junit.Test
+import org.junit.rules.TemporaryFolder
 import java.io.File
+import java.io.StringWriter
 
 class DependencyGraphGeneratorPluginTest {
+  @get:Rule val testProjectDir = TemporaryFolder()
+
   private lateinit var singleProject: DefaultProject
 
   @Before fun setUp() {
@@ -33,5 +39,62 @@ class DependencyGraphGeneratorPluginTest {
     assertThat(task.inputFile).hasToString(singleProject.buildFile.toString())
     assertThat(task.outputFileDot).hasToString(File(singleProject.buildDir, "dependency-graph.dot").toString())
     assertThat(task.outputFileImage).hasToString(File(singleProject.projectDir, "dependency-graph.png").toString())
+  }
+
+  @Test fun integrationTest() {
+    val classpathString = javaClass.classLoader.getResource("plugin-classpath.txt")
+        .readText()
+        .split(Regex("\\s*\\r?\\n\\s*"))
+        .map { File(it).absolutePath.replace("\\", "\\\\") }
+        .joinToString(", ") { "'$it'" }
+
+    val buildFile = testProjectDir.newFile("build.gradle")
+    buildFile.writeText("""
+        |buildscript {
+        |  dependencies {
+        |    classpath files($classpathString)
+        |  }
+        |}
+        |
+        |apply plugin: "java-library"
+        |apply plugin: "com.vanniktech.dependency.graph.generator"
+        |
+        |repositories {
+        |  mavenCentral()
+        |}
+        |
+        |dependencies {
+        |  api "org.jetbrains.kotlin:kotlin-stdlib:1.2.30"
+        |  implementation "io.reactivex.rxjava2:rxjava:2.1.10"
+        |}""".trimMargin())
+
+    val stdErrorWriter = StringWriter()
+
+    GradleRunner.create()
+        .withPluginClasspath()
+        .withGradleVersion("4.6")
+        .withProjectDir(testProjectDir.root)
+        .withArguments("generateDependencyGraph")
+        .forwardStdError(stdErrorWriter)
+        .build()
+
+    // No errors.
+    assertThat(stdErrorWriter).hasToString("")
+
+    // We don't want to assert the content of the image, just that it exists.
+    assertThat(File(testProjectDir.root, "dependency-graph.png")).exists()
+
+    assertThat(File(testProjectDir.root, "build/dependency-graph.dot")).hasContent("""
+        |digraph G {
+        |  ${testProjectDir.root.name} [label="${testProjectDir.root.name}", shape="box"];
+        |  orgjetbrainskotlinkotlinstdlib [label="kotlin-stdlib", shape="box"];
+        |  ${testProjectDir.root.name} -> orgjetbrainskotlinkotlinstdlib;
+        |  orgjetbrainsannotations [label="annotations", shape="box"];
+        |  orgjetbrainskotlinkotlinstdlib -> orgjetbrainsannotations;
+        |  ioreactivexrxjava2rxjava [label="rxjava", shape="box"];
+        |  ${testProjectDir.root.name} -> ioreactivexrxjava2rxjava;
+        |  orgreactivestreamsreactivestreams [label="reactive-streams", shape="box"];
+        |  ioreactivexrxjava2rxjava -> orgreactivestreamsreactivestreams;
+        |}""".trimMargin())
   }
 }

